packages/cardano-services/src/ChainHistory/DbSyncChainHistory/mappers.ts

```diff
@@ -105,14 +105,33 @@ export const mapWithdrawal = (withdrawalModel: WithdrawalModel): Cardano.Withdra
 // Remove this and select the actual redeemer data from `redeemer_data` table.
 const stubRedeemerData = Buffer.from('not implemented');
 
+const mapRedeemerPurpose = (purpose: string): Cardano.RedeemerPurpose => {
+  switch (purpose) {
+    case 'cert':
+      return Cardano.RedeemerPurpose.certificate;
+    case 'mint':
+      return Cardano.RedeemerPurpose.mint;
+    case 'spend':
+      return Cardano.RedeemerPurpose.spend;
+    case 'reward':
+      return Cardano.RedeemerPurpose.withdrawal;
+    case 'voting':
+      return Cardano.RedeemerPurpose.vote;
+    case 'proposing':
+      return Cardano.RedeemerPurpose.propose;
+    default:
```

Comment:
Consider doing something like this:
```ts
const redeemerToPurpose = {
  cert: Cardano.Redeemer.Purpose.certificate,
  mint: Cardano.RedeemerPurpose.mint,
  ....
}

const mapRedeemerPurpose = (purpose: string): Cardano.RedeemerPurpose => {
  return redeemerToPurpose[purpose] ?? throw NotImplementedError(...)
}

Reply:
Thanks for suggestion, I like the object approach :) https://github.com/input-output-hk/cardano-js-sdk/pull/1176/commits/bded200076767b3e38eba63c055b3b52a9f9b9fc